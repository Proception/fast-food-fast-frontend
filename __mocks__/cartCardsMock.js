import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../src/actions/orderActions';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [
      {
        "id": 1,
        "imageUrl": yam,
        "price": 500,
        "title": "Boiled Yam",
      },
      {
        "id": 2,
        "imageUrl": rice,
        "price": 900,
        "title": "Rice",
      },
      {
        "id": 3,
        "imageUrl": fufu,
        "price": 500,
        "title": "Fufu",
      },
    ],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: true,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};

const emptyCartProps = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: (e) => {},
  onChange: () => {},
  order: {
    cart: [],
    cartTotalPrice: 0
  },
  auth: {
    isLoading: false,
    user: {
      id: 1
    }
  },
  removeItemFromCart,
  updateCartTotal,
  updateItemQuantity,
  makeOrder
};
export { props, emptyCartProps };
  