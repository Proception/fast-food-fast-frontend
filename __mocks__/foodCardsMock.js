import yam from '../assets/images/yam.jpg';
import rice from '../assets/images/rice.jpg';
import fufu from '../assets/images/fufu.jpg';

const props = {
  item: [
    {
      "id": 1,
      "imageUrl": yam,
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": rice,
      "price": 900,
      "title": "Rice",
    },
    {
      "id": 3,
      "imageUrl": fufu,
      "price": 500,
      "title": "Fufu",
    },
  ],
  onClick: () => {},
  onChange: () => {}
};
export default props;
  