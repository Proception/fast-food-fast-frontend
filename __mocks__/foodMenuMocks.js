import { addItemToCart, updateCartTotal } from '../src/actions/orderActions';

const props = {
    addItemToCart,
    updateCartTotal,
    auth: {
      user: {
        id:1
      }
    }
    };
const emptyOrderProps = {
    updateCartTotal,
    addItemToCart,
    auth: {
    }
  };
export { props, emptyOrderProps };