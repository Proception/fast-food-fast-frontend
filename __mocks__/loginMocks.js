import { loginUser } from '../src/actions/authActions'

const props = {
    loginUser,
    auth: {
        isLoading: false,
    },
  };
  const propsIsLoading = {
    loginUser,
    auth: {
        isLoading: true,
    },
  };
  const state = {
    loginData: {
        email: 'omasan.esimaje',
        password: 'pass',
    }
  };
export { props, state, propsIsLoading };

