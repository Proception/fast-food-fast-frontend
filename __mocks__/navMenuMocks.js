import { setToken } from '../src/actions/authActions';

const stateGuest = {
    data: [],
    auth: {
        isAuthenticated: false,
    }
  };
  const stateUser = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
      }
    },
    setToken: setToken,
    order: {
      cart: [1,2]
    }
  };
  const stateUserEmptyCart = {
    data: [],
    auth: {
        isAuthenticated: true,
        user: {
          email: "omasan.esimaje@gmail.com",
          // full_name: "benedict"
          role_id: 200
      }
    },
    setToken: setToken,
    order: {
      cart: []
    }
  };
  const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwiZnVsbF9uYW1lIjoiQmVuZWRpY3QiLCJyb2xlX2lkIjoyMDAsImlhdCI6MTU1MTg3NzMyNSwiZXhwIjoxNTUxOTYzNzI1fQ.DwFFvZskGl9JeBnLD1CCnBSpYA1i0KgK1btUY1eLc9c'
  export { stateGuest, stateUser, token, stateUserEmptyCart };