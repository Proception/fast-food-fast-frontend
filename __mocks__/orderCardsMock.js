import { getOrders, getUserOrders } from '../src/actions/orderActions';
const props = {
    orders: [
        {
          "id": 1,
          "orderNo": "12",
          "price": 5000,
          "orderDate": "30 August, 2016",
          "orderStatus": "Pending",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 2,
          "orderNo": "54",
          "price": 1200,
          "orderDate": "Boiled Yam",
          "orderStatus": "In Progress",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 3,
          "orderNo": "67",
          "price": 1500,
          "orderDate": "30 October, 2017",
          "orderStatus": "Fulfilled",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
        {
          "id": 4,
          "orderNo": "33",
          "price": 3000,
          "orderDate": "30 January, 2018",
          "orderStatus": "Completed",
          "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
        },
      ],
    order: {
      orders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },],
      userOrders: [{
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },]
    },
    getOrders,
    getUserOrders
    };
const emptyOrderProps = {
  orders: [
      {
        "id": 1,
        "orderNo": "12",
        "price": 5000,
        "orderDate": "30 August, 2016",
        "orderStatus": "Pending",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 2,
        "orderNo": "54",
        "price": 1200,
        "orderDate": "Boiled Yam",
        "orderStatus": "In Progress",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 3,
        "orderNo": "67",
        "price": 1500,
        "orderDate": "30 October, 2017",
        "orderStatus": "Fulfilled",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
      {
        "id": 4,
        "orderNo": "33",
        "price": 3000,
        "orderDate": "30 January, 2018",
        "orderStatus": "Completed",
        "orderAddress": "52, Enitan str, Aguda, Surulere, Lagos State",
      },
    ],
  order: {
    orders: [],
    userOrders: [],
  },
  getOrders,
  getUserOrders
  };
export { props, emptyOrderProps };