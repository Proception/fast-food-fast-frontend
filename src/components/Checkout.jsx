import React, { Component } from 'react';
import { connect } from 'react-redux';
import CheckoutCards from './CheckoutCards.jsx';
import { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder } from '../actions/orderActions';
import Notify from '../utils/Notify';

export class Checkout extends Component {
  
  constructor(props) {
    super(props);
    this.handleOrder = this.handleOrder.bind(this);
    this.onChange = this.onChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick = (e, data) => {
    const { removeItemFromCart, updateCartTotal } = this.props;
    removeItemFromCart(data);
    updateCartTotal();
  }

  onChange(e, data) {
    const {id, value } = e.target;
    const { updateItemQuantity, updateCartTotal } = this.props;
    updateItemQuantity({id, value});
    updateCartTotal();
  }

  handleOrder() {
    const { cartTotalPrice, cart } = this.props.order;
    const { makeOrder } = this.props;
    const menuItems = [];
    cart.forEach(item => {
      menuItems.push(
        {
          menuId: item.id,
          quantity: item.quantity
        }
      )
    });

    if (!Array.isArray(menuItems) || menuItems.length > 0){
      makeOrder({
        userId: this.props.auth.user.id,
        orderAmount: cartTotalPrice,
        menu: [...menuItems]
        });
        Notify.notifySuccess('Checkout Successful');
    }

    if (menuItems.length === 0) {
      Notify.notifyError('Cart is empty');
    }
  }

  generateCart() {
    const { cart } = this.props.order;
    return (
      <div id="sec">
        {cart.map(item => (
          <CheckoutCards onChange={this.onChange} onClick={ ((e) => this.handleClick(e, item)) } item={item} />
        ))}
      </div>
    )
  }
  render() {
    const { cartTotalPrice } = this.props.order;
    return (
      <section className="order">
        <div className="container">
          <div className="fullsection">
            <h3>Your Food Selection</h3>
            <div className="checkout">
              <p><label htmlFor="price">Sum Total : </label><span id="total">&#8358;{cartTotalPrice}</span></p>
              <p><input 
                onClick={this.handleOrder } 
                id="checkout" className="button" type="button" name="" 
                value={this.props.auth.isLoading ? 'Processing' : 'Checkout'} /></p>
            </div>
            {
              this.generateCart()
            }
          </div>
        </div>
      </section>
    )
  }
}

const mapStateToProps = state => ({
  auth: state.auth,
  order: state.order,
});

export default connect(mapStateToProps, { removeItemFromCart, updateCartTotal, updateItemQuantity, makeOrder })(Checkout);