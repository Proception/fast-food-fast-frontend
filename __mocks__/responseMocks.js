import { token } from './navMenuMocks';

export const okResponse = {
    status: 200,
    response: {
        code: 200,
        status: 'ok',
        messages: 'User logged in successfully',
        data: token
    }
};

export const unauthorizedResponse = {
    status: 401,
    response: {
        code: 401,
        status: 'ok',
        messages: 'User is not authorized',
        data: token
    }
};

export const createdResponse = {
    status: 201,
    response: {
        code: 201,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const noContentResponse = {
    status: 204,
    response: {
        code: 204,
        status: 'Created',
        messages: 'User signed up in successfully',
        data: token
    }
};

export const conflictResponse = {
    status: 409,
    response: {
        code: 409,
        status: 'Conflict',
        messages: 'User exists on the platform',
        data: token
    }
}

export const badRequestResponse = {
    status: 400,
    response: {
        code: 400,
        status: 'Bad Request',
        messages: 'Invalid credentials',
        data: token
    }
}
