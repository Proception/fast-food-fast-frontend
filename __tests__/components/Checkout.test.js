import React from 'react';
import { shallow } from 'enzyme';
import { Checkout } from '../../src/components/Checkout.jsx';
import { props, emptyCartProps } from '../../__mocks__/cartCardsMock';
import {   preventDefault } from '../../__mocks__/loginMocks';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Checkout { ...props }/>);
});
describe('Checkout Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('calls handleClick(e)', () => {
    const data = {
      "id": 1,
      "imageUrl": 'yam',
      "price": 500,
      "title": "Boiled Yam",
    };
    sinon.spy(wrapper.instance(), 'handleClick');

    wrapper.instance().handleClick(preventDefault, data);

    expect(wrapper.instance().handleClick.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleClick.calledWith(preventDefault, data));
  });
  it('calls onChange(e)', () => {
    const data = {
      "id": 1,
      "imageUrl": 'yam',
      "price": 500,
      "title": "Boiled Yam",
    };
    const e = {
      target: {
        id: 1,
        value: 10,
      }
    } 
    sinon.spy(wrapper.instance(), 'onChange');

    wrapper.instance().onChange(e, data);

    expect(wrapper.instance().onChange.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().onChange.calledWith(e, data));
  });
  it('calls handleOrder(e)', () => {
    const cart = [{
      "id": 1,
      "imageUrl": 'yam',
      "price": 500,
      "title": "Boiled Yam",
    },
    {
      "id": 2,
      "imageUrl": 'rice',
      "price": 900,
      "title": "Rice",
    }];
    const totalCost = 500;
    sinon.spy(wrapper.instance(), 'handleOrder');

    wrapper.instance().handleOrder(preventDefault, cart, totalCost);

    expect(wrapper.instance().handleOrder.calledOnce)
      .toEqual(true);
    expect(wrapper.instance().handleOrder.calledWith(preventDefault, cart, totalCost));
  });
  it('calls handleOrder(e) empty cart', () => {
    wrapper = shallow(<Checkout { ...emptyCartProps }/>);
    sinon.spy(wrapper.instance(), 'handleOrder');

    wrapper.instance().handleOrder();

    expect(wrapper.instance().handleOrder.calledOnce)
      .toEqual(true);
  });
  it('calls generateCart empty cart', () => {
    wrapper = shallow(<Checkout { ...emptyCartProps }/>);
    sinon.spy(wrapper.instance(), 'generateCart');

    wrapper.instance().generateCart();

    expect(wrapper.instance().generateCart.calledOnce)
      .toEqual(true);
  });

});
