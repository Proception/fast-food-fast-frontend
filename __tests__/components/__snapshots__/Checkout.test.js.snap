// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Component should match snapshot 1`] = `
<section
  className="order"
>
  <div
    className="container"
  >
    <div
      className="fullsection"
    >
      <h3>
        Your Food Selection
      </h3>
      <div
        className="checkout"
      >
        <p>
          <label
            htmlFor="price"
          >
            Sum Total : 
          </label>
          <span
            id="total"
          >
            â‚¦
            0
          </span>
        </p>
        <p>
          <input
            className="button"
            id="checkout"
            name=""
            onClick={[Function]}
            type="button"
            value="Processing"
          />
        </p>
      </div>
      <div
        id="sec"
      >
        <CheckoutCards
          item={
            Object {
              "id": 1,
              "imageUrl": "test-file-stub",
              "price": 500,
              "title": "Boiled Yam",
            }
          }
          onChange={[Function]}
          onClick={[Function]}
        />
        <CheckoutCards
          item={
            Object {
              "id": 2,
              "imageUrl": "test-file-stub",
              "price": 900,
              "title": "Rice",
            }
          }
          onChange={[Function]}
          onClick={[Function]}
        />
        <CheckoutCards
          item={
            Object {
              "id": 3,
              "imageUrl": "test-file-stub",
              "price": 500,
              "title": "Fufu",
            }
          }
          onChange={[Function]}
          onClick={[Function]}
        />
      </div>
    </div>
  </div>
</section>
`;
